netket/vmc_common.py

```diff
@@ -0,0 +1,92 @@
+import sys
+
+import netket as _nk
+
+
+def info(obj, depth=None):
+    if hasattr(obj, "info"):
+        return obj.info(depth)
+    else:
+        return str(obj)
+
+
+def make_optimizer_fn(arg, ma):
```

Comment:
Just for future reference, this should be completely removed in favor of an AbstractOptimizer class in python and possibly a wrapper for JAX/Tensorflow/Pytorch optimizers, as done for machines now